packages/terra-hookshot/src/_HookshotUtils.js

```diff
@@ -0,0 +1,537 @@
+ *
+ * @ param {Object} tRect - The target rectangle..
+ * @ param {Object} tAttachment - The vertical and horizonal hookshot attachments of the target.
+ * @ param {Object} tOffset - The vertical and horizonal offset values of the target.
+ * @ param {Object} cAttachment - The vertical and horizonal hookshot attachments of the content.
+ */
+const mirrorTargetCoords = (tRect, tAttachment, tOffset, cAttachment) => {
+  const mOffset = mirrorOffset(tOffset, tAttachment);
+  let mAttachment = tAttachment;
+  if (cAttachment.vertical === tAttachment.vertical || (cAttachment.vertical !== 'middle' && tAttachment.vertical !== 'middle')) {
+    mAttachment = mirrorAttachment(tAttachment);
+  }
+  return getTargetCoords(tRect, mAttachment, mOffset);
+};
+
+/**
+ * The function returns the unadjusted content positional value.
+ *
+ * @ param {Object} rect - The content rectangle.
+ * @ param {Object} attachment - The vertical and horizonal hookshot attachments of the content.
+ * @ param {Object} offset - The content offset value.
+ * @ param {Object} tCoords - The screen coordinates of the target.
+ * @ param {number} margin - The px value of the attachmentMargin.
+ */
+const getBasicPositions = (rect, attachment, offset, tCoords, margin) => {
+  const cCoords = {};
+  if (attachment.vertical === 'middle') {
+    if (attachment.horizontal === 'center') {
+      cCoords.x = tCoords.x - (rect.width / 2);
+    } else if (attachment.horizontal === 'right') {
+      cCoords.x = tCoords.x - rect.width - margin;
+    } else {
+      cCoords.x = tCoords.x + margin;
+    }
+
+    cCoords.y = tCoords.y - (rect.height / 2);
+  } else {
+    if (attachment.horizontal === 'center') {
+      cCoords.x = tCoords.x - (rect.width / 2);
+    } else if (attachment.horizontal === 'right') {
+      cCoords.x = tCoords.x - rect.width;
+    } else {
+      cCoords.x = tCoords.x;
+    }
+
+    if (attachment.vertical === 'bottom') {
+      cCoords.y = tCoords.y - rect.height - margin;
+    } else {
+      cCoords.y = tCoords.y + margin;
+    }
+  }
+
+  return {
+    cCoords: {
+      x: cCoords.x + offset.horizontal,
+      y: cCoords.y + offset.vertical,
+      attachment,
+      offset,
+    },
+    tCoords,
+  };
+};
+
+/**
+ * This method returns parent elements capable of scrolling until a fixed element or document is found.
+ *
+ * @ param {Object} positions - The basic positional values to be evaluated.
+ * @ param {Object} cRect - The content rectangle.
+ * @ param {Object} bRect - The bounding rectangle.
+ * @ param {Object} tRect - The target rectangle.
+ * @ param {number} margin - The px value of the attachmentMargin.
+ * @ param {string} behavior - The fallback pattern for available rotations.
+ */
+const getRotatedPositions = (positions, cRect, bRect, tRect, margin, behavior) => {
+  if (isValidPositions(positions, cRect, bRect)) {
+    return positions; // default valid
+  }
+
+  if (behavior !== 'none') {
```

Comment:
The logic in this method seems like it could be simplified. I'm thinking something like 
```
if (behavior === 'none') {
   return positions;
}  else if ( behavior === 'flip') {
// get new positions
} else if ( behavior === 'auto') {
// get new positions
}

return isValidPositions() ? newPositions : positions;
```

Reply:
I'm trying to think of a way to trim it down, but it's a fallback pattern:
Primary>180>90>-90>Primary
So each step needs to be validated. Auto includes flip as well.